core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -66,8 +78,45 @@
     public final static String INDEX = ".ingest";
     public final static String TYPE = "pipeline";
 
+    final static Settings INGEST_INDEX_SETTING = Settings.builder()
+        .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+        .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+        .put("index.mapper.dynamic", false)
+        .build();
+
+    final static String PIPELINE_MAPPING;
+
+    static {
+        try {
+            PIPELINE_MAPPING = XContentFactory.jsonBuilder().startObject()
+                .field("dynamic", "strict")
+                .startObject("_all")
+                    .field("enabled", false)
+                .endObject()
+                .startObject("properties")
+                    .startObject("processors")
```

Comment:
I don't see how the cluster state is not *always* the right place to store configuration. Putting it in an index is a hack. I don't see the issue with "large pipeline configuration", but I also don't see how it would differ from how cluster state for a specific index is passed where necessary, eg mappings are not loaded for all indexes on all nodes.